src/Spice86.Core/Emulator/OperatingSystem/DosPathResolver.cs

```diff
@@ -0,0 +1,264 @@
+            !IsWithinMountPoint(hostFullPath, _driveMap[driveLetter]) ||
+            Encoding.ASCII.GetByteCount(hostFullPath) > 255) {
+            return DosFileOperationResult.Error(ErrorCode.PathNotFound);
+        }
+
+        _driveMap[driveLetter].CurrentDirectory = hostFullPath;
+        return DosFileOperationResult.NoValue();
+    }
+
+    /// <summary>
+    /// Converts the DOS path to a full host path of the parent directory.<br/>
+    /// </summary>
+    /// <param name="dosPath">The DOS path to convert.</param>
+    /// <returns>A string containing the full path to the parent directory in the host file system, or <c>null</c> if nothing was found.</returns>
+    public string? TryGetFullHostParentPathFromDos(string dosPath) {
+        string? fullHostPath = TryGetFullHostPathFromDos(dosPath);
+        if(string.IsNullOrWhiteSpace(fullHostPath)) {
+            return null;
+        }
+        string? parent = Directory.GetParent(fullHostPath)?.FullName;
+        if (string.IsNullOrWhiteSpace(parent)) {
+            return null;
+        }
+
+        return ConvertUtils.ToSlashPath(parent);
+    }
+
+    private (string HostPrefixPath, string DosRelativePath) DeconstructDosPath(string dosPath) {
+        if (IsPathRooted(dosPath)) {
+            int length = dosPath.TakeWhile(x => x == '\\' || x == '/').Count();
+            if (StartsWithDosDrive(dosPath)) {
+                length = 3;
+            }
+            return (_driveMap[_currentDrive].MountPoint, dosPath[length..]);
+        } else if (StartsWithDosDrive(dosPath)) {
+            return (_driveMap[dosPath[0]].MountPoint, dosPath[2..]);
+        } else {
+            return (_driveMap[_currentDrive].MountPoint, dosPath);
+        }
+    }
+
+    /// <summary>
+    /// Converts the DOS path to a full host path.<br/>
+    /// </summary>
+    /// <param name="dosPath">The DOS path to convert.</param>
+    /// <returns>A string containing the full file path in the host file system, or <c>null</c> if nothing was found.</returns>
+
+    public string? TryGetFullHostPathFromDos(string dosPath) {
+        if(string.IsNullOrWhiteSpace(dosPath)) {
+            return null;
+        }
+        (string HostPrefix, string DosRelativePath) = DeconstructDosPath(dosPath);
+
+        if(string.IsNullOrWhiteSpace(DosRelativePath)) {
+            return ConvertUtils.ToSlashPath(HostPrefix);
+        }
+
+        DirectoryInfo hostDirInfo = new DirectoryInfo(HostPrefix);
+
+        string? relativeHostPath = hostDirInfo
+            .EnumerateDirectories("*", new EnumerationOptions() {
+                RecurseSubdirectories = true,
+            })
+            .Select(x => (FileSystemInfo)x)
+            .Concat(
+            hostDirInfo.EnumerateFiles("*",
+            new EnumerationOptions() {
+                RecurseSubdirectories = true,
+            }).Select(x => x))
+            .FirstOrDefault(x =>
+                x is FileInfo ?
```

Comment:
Is it possible to move the lambda to a separate method? The ternary is long and makes my head hurt :)

Reply:
Done.